### DISCUSSION TITLE
Please consider switching from udisks2 to lsblk

### DISCUSSION THREAD

**Discussion by gardar** [fwupd/fwupd#7320] [General]
**Is your feature request related to a problem? Please describe.**
`fwupd` currently relies on `udisks2` for certain disk management tasks, such as detecting the ESP (EFI System Partition). This introduces additional dependencies and overhead. On lightweight systems, `udisks2` pulls in a lot of dependencies and requires D-Bus, which may not be necessary for other applications.

**Describe the solution you'd like**
Would it be possible to switch from `udisks2` to `lsblk` for detecting the ESP partition? `lsblk` can provide the necessary block device information without the complexity and overhead associated with `udisks2`, potentially reducing the dependency chain on lightweight systems.

**Describe alternatives you've considered**

**Additional context**
`lsblk` is part of the `util-linux` package and is included by default in most Linux distributions. While `udisks2` might be used for other tasks within `fwupd`, starting with the switch for ESP detection could simplify the dependency chain and improve performance on lightweight systems by eliminating the need for D-Bus and unnecessary dependencies.

**Comment by hughsie** [comment]
Does lsblk provide a DBus or shared library API we can query? We're using quite a bit of functionality from udisks now.

**Reply by gardar** [reply]
`lsblk` is primarily a command line utility with advanced query capabilities and does have a lot of different output formats for scripting purposes (raw/json/etc). https://github.com/util-linux/util-linux/blob/master/misc-utils/lsblk.8.adoc

I don't believe it has d-bus capabilities but perhaps https://github.com/util-linux/util-linux/blob/master/misc-utils/lsblk.h can be of use?

**Reply by gardar** [reply]
Just to give an example, here's how you can find the mountpoints of the esp partitions: `lsblk -lno mountpoint --filter 'PARTTYPENAME=="EFI System"'`

**Comment by superm1** [comment]
So we use udisks2 for more than just identifying the ESP.  We identify free space on partitions, we mount/unmount, prioritize different ESP's against each other, look for MD arrays, and miscategorized partitions.

I don't think it's trivial to just shell out to `lsblk` for this.  If you really want support for it go ahead and code it up, I'm happy to review a pull request.  IMO The way it would be done is an abstraction in FuVolume for different "backends".  So udisks2 can be a backend for disk access and then you can have another backend that uses lsblk and other tools.
